### PR TITLE
Integrate golangci-lint to run aggregate linting checks for CI

### PR DESCRIPTION
**Pull request by timflannagan**
**Description of the change:**

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+run:
+  timeout: 5m
+  skip-dirs:
+  - pkg/lib
+  - pkg/api
+  - pkg/fakes
+  - pkg/package-server/apis
+  - test/e2e
```
Comment (timflannagan) [inline .golangci.yaml:8]: Removing this can be done in a follow-up PR.

File: .golangci.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  - gofmt
+  - goimports
+  - importas
+  - misspell
+  - stylecheck
+  - tparallel
+  - unconvert
+  - whitespace
+  disable:
+  - errcheck
```
Comment (timflannagan) [inline .golangci.yaml:22]: Removing this can be done in a follow-up PR.

File: .golangci.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+issues:
+  max-issues-per-linter: 0
+  max-same-issues: 0
```
Comment (timflannagan) [inline .golangci.yaml:26]: Setting both of these configuration options limit the number of linting violations that are discovered at a time. Setting the values to `0` disables that behavior.

File: cmd/catalog/main.go [hunk 4]
```diff
@@ -69,7 +69,7 @@ var (
 	tlsCertPath = flag.String(
 		"tls-cert", "", "Path to use for certificate key (requires tls-key)")
 
-	profiling = flag.Bool("profiling", false, "deprecated")
+	_ = flag.Bool("profiling", false, "deprecated")
```
Comment (timflannagan) [inline cmd/catalog/main.go:72]: This flag is currently unused right now since we merge the profiling/metrics/health/etc. endpoints last minor release. I thought this was the easiest solution, vs. updating the .golangci-lint.yaml configuration and explicitly allowlisting this flag.

File: pkg/controller/operators/olm/operator.go [hunk 5]
```diff
@@ -1718,6 +1712,10 @@ func (a *Operator) transitionCSVState(in v1alpha1.ClusterServiceVersion) (out *v
 
 	// Check for intersecting provided APIs in intersecting OperatorGroups
 	allGroups, err := a.lister.OperatorsV1().OperatorGroupLister().List(labels.Everything())
+	if err != nil {
+		logger.WithError(err).Warn("failed to list operatorgroups")
+		return
+	}
```
Comment (timflannagan) [inline pkg/controller/operators/olm/operator.go:1718]: Note: need to double-check whether this is the correct behavior. It wasn't clear to me poking around whether I should be returning a syncError here.

File: .golangci.yaml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  - pkg/fakes
+  - pkg/package-server/apis
+  - test/e2e
+
+linters:
+  enable:
+  - depguard
+  - gofmt
+  - goimports
+  - importas
```
Comment (timflannagan) [inline .golangci.yaml:15]: I'd like to tackle adding a more fleshed out importas configuration ( e.g. enforcing consistency with corev1/metav1/operatorsv1alpha1/etc. package alias') as a follow-up.

**Review by dinhxuanvu — COMMENTED** [review]
Looking good. Just a quick question.

File: .golangci.yaml (hunk 1 above)
Comment (dinhxuanvu) [reply .golangci.yaml:8]: Can you explain the exclusion for these paths? I can see why we don't care about fakes and test but the rest are good to have.

Comment (timflannagan) [reply .golangci.yaml:8]: Yeah, definitely - I had a couple of reasons:

- Aim to reduce the reviewer burden with an already large changeset;
- Get the main infrastructure in place so we golangci-lint can identify linting issues ahead of time to reduce reviewer responsibilities going forward vs. fixing all lints all at once;
- I was already fairly burnt out identifying and manually fixing all the linting violations and didn't want to hold this up further, as I'm less concerned about test code

**Review by dinhxuanvu — APPROVED** [review]
/approve

File: pkg/controller/operators/catalog/subscription/state.go [hunk 7]
```diff
@@ -243,7 +243,6 @@ func (c *catalogHealthState) UpdateHealth(now *metav1.Time, client clientv1alpha
 	var known CatalogHealthKnownState
 	switch {
 	case missingTargeted:
-		healthy = false
```
Comment (awgreene) [inline pkg/controller/operators/catalog/subscription/state.go:246]: Why was this removed?

Comment (awgreene) [reply pkg/controller/operators/catalog/subscription/state.go:246]: Ah, it isn't used anywhere below.

**Review by awgreene — APPROVED** [review]
/approve

Great work on this @timflannagan
